code/game/machinery/buttons.dm

```diff
@@ -56,9 +56,6 @@
 			icon_state = skin
 
 /obj/machinery/button/attackby(obj/item/W, mob/user, params)
-	if(istype(W, /obj/item/device/detective_scanner))
-		return

```

Comment:
There will be plenty of removal like this in the PR. No more hacky istype checks, we now make sure attackby returns 1 so as to not call the item's afterattack.